compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeNarrower.java

```diff
@@ -368,7 +313,7 @@ private NarrowedTypes getNarrowedTypesForBinaryOp(Map<BVarSymbol, NarrowedTypes>
             lhsTrueType = narrowedTypes.trueType;
             lhsFalseType = narrowedTypes.falseType;
         } else {
-            lhsTrueType = lhsFalseType = symbol.type;
```

Comment:
Can't we fix where the symbol is passed instead? To use the narrowed shadowed one?